odl/discr/partition.py

```diff
@@ -766,6 +701,202 @@ def uniform_partition_fromgrid(grid, begin=None, end=None):
     return RectPartition(IntervalProd(begin_vec, end_vec), grid)
 
 
+def uniform_partition(begin=None, end=None, num_nodes=None,
+                      cell_sides=None, nodes_on_bdry=False):
+    """Return a partition with equally sized cells.
+
+    This function offers more flexibility over `uniform_partition` in
+    that any combination of 3 or 4 out of 4 possible init parameters
+    can be given, and the (possibly) missing one is calculated.
+
+    Parameters
+    ----------
+    begin, end : float or array-like, optional
+        Vectors defining the begin end end points of an `IntervalProd`
+        (a rectangular box). For one-dimensional partitions, single
+        floats can be provided. ``None`` entries mean "compute the value".
+    num_nodes : int or sequence of int, optional
+        Number of nodes per axis. For 1d intervals, a single integer
+        can be specified. ``None`` entries mean "compute the value".
+    cell_sides : float or array-like, optional
+        Side length of the partition cells per axis. For 1d intervals,
+        a single integer can be specified. ``None`` entries mean
+        "compute the value".
+    nodes_on_bdry : `bool` or `sequence`, optional
+        If a sequence is provided, it determines per axis whether to
+        place the last grid point on the boundary (True) or shift it
+        by half a cell size into the interior (False). In each axis,
+        an entry may consist in a single `bool` or a 2-tuple of
+        `bool`. In the latter case, the first tuple entry decides for
+        the left, the second for the right boundary. The length of the
+        sequence must be ``array.ndim``.
+
+        A single boolean is interpreted as a global choice for all
+        boundaries.
+
+    Notes
+    -----
+    In each axis, 3 of the 4 possible parameters must be given. If all
```

Comment:
Somewhat redundant given the introduction. Also, this now takes 5 parameters so we need to specify which 4.